src/ui/pages/SidePage/components/IncomingRequest/components/SignTransactionRequest.tsx

```diff
@@ -20,38 +20,30 @@ const SignTransactionRequest = ({
   handleCancel,
 }: RequestProps) => {
   const signTransaction = requestData.signTransaction;
+  const [isSigningObject, setIsSigningObject] = useState(false);
   const ballotLogo = requestData.logo ? requestData.logo : CardanoLogo;
 
   const signDetails = useMemo(() => {
     if (!signTransaction) return {};
 
-    return {
-      [`${i18n.t("request.signtransaction.transaction.action")}`]:
-        signTransaction.action,
-      [`${i18n.t("request.signtransaction.transaction.actionText")}`]:
-        signTransaction.actionText,
-      [`${i18n.t("request.signtransaction.transaction.id")}`]:
-        signTransaction.data.id,
-      [`${i18n.t("request.signtransaction.transaction.address")}`]:
-        signTransaction.data.address,
-      [`${i18n.t("request.signtransaction.transaction.event")}`]:
-        signTransaction.data.event,
-      [`${i18n.t("request.signtransaction.transaction.category")}`]:
-        signTransaction.data.category,
-      [`${i18n.t("request.signtransaction.transaction.proposal")}`]:
-        signTransaction.data.proposal,
-      [`${i18n.t("request.signtransaction.transaction.network")}`]:
-        signTransaction.data.network,
-      [`${i18n.t("request.signtransaction.transaction.votedat")}`]:
-        signTransaction.data.votedAt,
-      [`${i18n.t("request.signtransaction.transaction.votingpower")}`]:
-        signTransaction.data.votingPower,
-      [`${i18n.t("request.signtransaction.transaction.slot")}`]:
-        signTransaction.slot,
-      [`${i18n.t("request.signtransaction.transaction.uri")}`]:
-        signTransaction.uri,
-    };
-  }, [signTransaction]);
+    let signTransactionContent;
+    try {
+      signTransactionContent = JSON.parse(signTransaction.payload.payload);
+      setIsSigningObject(true);
+    } catch (error) {
+      signTransactionContent = signTransaction.payload.payload;
+    }
+    if (isSigningObject) {
```

Comment:
We just want to return the JSON. We should not be adding `signTransaction.type` to the JSON...

We need to show _exactly_ what the user is signing. (pretty print difference is OK, but not adding other fields)